csrgen.py

```diff
@@ -0,0 +1,190 @@
+                     help="Specify a domain or a file with list of domains")
+    csr.add_argument("-o", "--org", help="Organization Name")
+
+    config = subparser.add_parser("config", help="Create config file")
+    config.add_argument("-c", "--country", help="Country short code")
+    config.add_argument("-l", "--locality", help="Locality/City")
+    config.add_argument("-s", "--state", help="State Name")
+    config.add_argument("-o", "--org", help="Organization Name", required=True)
+
+    if parser.parse_args().command:
+        return parser.parse_args()
+    else:
+        parser.print_help()
+        exit()
+
+
+def gen_csr_config(args, config_file):
+    tm = env.get_template(config_file)
+    print(config_file)
+    SANS = list()
+    print("Generating CSR configuration..")
+    if type(args.domain) is tuple:
+        domain_file, is_file = args.domain
+        with open(domain_file) as fp:
+            domain = fp.readline()
+            count = 0
+            while domain:
+                print(domain)
+                if count == 0:
+                    CN = domain.strip().lower()
+                    domain = fp.readline()
+                    count += 1
+                elif 1 <= count <= 100:
+                    SANS.append(domain.strip().lower())
+                    domain = fp.readline()
+                    count += 1
+                else:
+                    print("Too many domains! Please limit to 100")
+                    exit(1)
+            print(CN)
+    else:
+        domain = args.domain
+        CN = domain.strip().lower()
+
+    generated_csr_config = open(args.domain + '_csr_config', 'w')
+    generated_csr_config.write(tm.render(cn=CN, sans=SANS))
+    generated_csr_config.close()
+
+    return str(args.domain + '_csr_config')
+
+
+def main(args):
+    if args.command == 'csr':
+        # Look for csrgen config files
+        config_file_search = fnmatch.filter(os.listdir(), "*_gen_config")
+        if len(config_file_search) == 0:
+            print("No config files found please run with 'config -h' to create one")
+            exit()
+        elif len(config_file_search) > 1:   # Check if there is more then one config file
+            if args.org:
+                # Check if any of the config files match the specified Org
+                if check_file({args.org + "_gen_config"}):
+                    csr_config_out = gen_csr_config(
+                        args, {args.org + "_gen_config"})
+            else:
+                print("Please specify Organization name with -o <Org>")
+                exit()
+        else:  # There is only one found so assume its the right one\
+            csr_config_out = gen_csr_config(args, config_file_search[0])
+
+        # Validate key argument and file
+        if args.key:
+            if check_file(args.key):
+                try:
+                    subprocess.call(['openssl', 'rsa', '-in',
+                                    args.key, '-check'])
```

Comment:
One nicer way to do this is
```suggestion
                    subprocess.call(' '.split(f"openssl rsa -in {args.key} -check"))
```

Reply:
Actually, yeah, it doesn't handle whitespace.